core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala

```diff
@@ -43,35 +43,53 @@ private[spark] class ShuffleMapStage(
     val shuffleDep: ShuffleDependency[_, _, _])
   extends Stage(id, rdd, numTasks, parents, firstJobId, callSite) {
 
+  private[this] var _mapStageJobs: List[ActiveJob] = Nil
```

Comment:
This is if multiple threads submit the same stage with submitMapStage. You're not bypassing the DAGScheduler when you do adaptive execution, you're just telling it to run a map stage even if there's no result stage downstream. But it still does all its standard stuff to figure out previous stages to launch, track dependencies when recovering from failures, etc.